app/controllers/article_controller.rb

```diff
@@ -191,43 +154,31 @@ def show
     session[:col_id] = @collection.slug
   end
 
-
   # display the article upload form
   def upload_form
   end
 
-
-  def find_or_create_category(collection, title)
```

Comment:
Was this method dead code?  It's strange to exist in the controller...

Reply:
This is not dead code (though it is untested). I merely moved it to private function as this is not a route. I suggest refactoring like this for all functions that are not routes in controller. We should be moving them to private functions

Reply:
Agreed completely.